apps/web/src/components/Posts/SharePostModal.tsx

```diff
@@ -0,0 +1,251 @@
+            tokens {
+              ...getPreviewImageUrlsInlineDangerouslyFragment
+            }
+          }
+        }
+      }
+    `,
+    { postId: postId }
+  );
+
+  const { post } = queryResponse;
+  const postUrl = `${getBaseUrl()}/post/${postId}`;
+
+  const { pushToast } = useToastActions();
+
+  const handleCopyToClipboard = useCallback(async () => {
+    void navigator.clipboard.writeText(postUrl);
+    pushToast({ message: 'Copied link to clipboard', autoClose: true });
+  }, [postUrl, pushToast]);
+
+  const handleShareButtonClick = useCallback(
+    (baseComposePostUrl: string) => {
+      const message = `I just posted ${tokenName} on gallery ${postUrl}`;
+      const encodedMessage = encodeURIComponent(message);
+
+      window.open(`${baseComposePostUrl}?text=${encodedMessage}`, '_blank');
+    },
+    [tokenName, postUrl]
+  );
+
+  // stripped down version of the pfp retrieving logic in ProfilePicture.tsx
+  const profileImageUrl = useMemo(() => {
+    if (!post || post?.__typename !== 'Post') {
+      return null;
+    }
+
+    const { token, profileImage } = post.author.profileImage ?? {};
+
+    if (profileImage && profileImage.previewURLs?.medium) {
+      return profileImage.previewURLs.medium;
+    }
+
+    if (token) {
+      const result = getPreviewImageUrlsInlineDangerously({
+        tokenRef: token,
+      });
+      if (result.type === 'valid') {
+        return result.urls.large;
+      }
+      return null;
+    }
+
+    return null;
+  }, [post]);
+
+  if (post?.__typename !== 'Post') {
+    return null;
+  }
+
+  const token = post.tokens?.[0];
+  if (!token) {
+    return null;
+  }
+
+  const result = getPreviewImageUrlsInlineDangerously({ tokenRef: token });
+  if (result.type !== 'valid') {
+    return null;
+  }
+
+  const shareButtonsDetails = [
+    {
+      icon: <FarcasterIcon fill="#FEFEFE" />,
+      title: 'WARPCAST',
+      baseComposePostUrl: 'https://warpcast.com/~/compose',
+    },
+    {
+      icon: <LensIcon fill="#FEFEFE" />,
+      title: 'LENS',
+      baseComposePostUrl: 'https://hey.xyz/',
+    },
+    {
+      icon: <TwitterIcon fill="#FEFEFE" />,
+      title: 'TWITTER',
+      baseComposePostUrl: 'https://twitter.com/intent/tweet',
+    },
+  ];
+
+  const imageUrl = result.urls.small ?? '';
+  const username = post.author.username ?? '';
+  const caption = post.caption ?? '';
+
+  return (
+    <StyledContainer>
+      <VStack gap={16}>
+        <HStack>
+          <MiniPostOpenGraphPreview
+            caption={caption}
+            username={username}
+            imageUrl={imageUrl}
+            profileImageUrl={profileImageUrl ?? ''}
+          />
+        </HStack>
+        <HStack gap={8}>
+          {shareButtonsDetails.map((btnData) => (
+            <ShareButton
+              key={btnData.title}
+              title={btnData.title}
+              icon={btnData.icon}
+              onClick={() => handleShareButtonClick(btnData.baseComposePostUrl)}
+            />
```

Comment:
we should track specifically which platform is being used. will push up the change!

Reply:
oh right. yess, will make sure!